File: internal/tools/scraping/scraping.go
package scraping

import (
	"encoding/json"
	"fmt"

	"github.com/go-resty/resty/v2"
)

type ScrapingTool struct{}

func NewScrapingTool() *ScrapingTool {
	return &ScrapingTool{}
}

type ScrapingArguments struct {
	Url string `json:"url"`
}

func (s *ScrapingTool) CallTool(arguments string) string {
	var args ScrapingArguments
	if err := json.Unmarshal([]byte(arguments), &args); err != nil {
		return fmt.Sprintf("Error parsing arguments: %v", err)
	}

	apiUrl := fmt.Sprintf("https://r.jina.ai/%s", args.Url)

	client := resty.New()

	resp, err := client.R().Get(apiUrl)
	if err != nil {
		return fmt.Sprintf("Error making request: %v", err)
	}

	if resp.StatusCode() != 200 {
		return fmt.Sprintf("Error response from API: %s", resp.Status())
	}

	return resp.String()
}

File: .teo/skills/scraping/scripts/scraping.py
import json
import sys
import urllib.request
import urllib.error

def main():
    if len(sys.argv) < 2:
        print("Error: No arguments provided")
        sys.exit(1)

    try:
        params = json.loads(sys.argv[1])
    except json.JSONDecodeError:
        print("Error: Invalid JSON argument")
        sys.exit(1)

    url_arg = params.get("url")
    if not url_arg:
        print("Error: url is required")
        sys.exit(1)

    # Jina AI Reader API
    api_url = f"https://r.jina.ai/{url_arg}"
    
    try:
        req = urllib.request.Request(api_url)
        # Jina AI might require headers sometimes, but usually works without for basic usage
        # Adding a User-Agent just in case
        req.add_header('User-Agent', 'Mozilla/5.0 (compatible; TeoSkill/1.0)')

        with urllib.request.urlopen(req) as response:
            if response.status != 200:
                print(f"Error response from API: {response.status}")
                return
            
            print(response.read().decode('utf-8'))

    except urllib.error.HTTPError as e:
         print(f"Error making request: {e}")
    except Exception as e:
        print(f"Error making request: {e}")

if __name__ == "__main__":
    main()
